scio-core/src/main/scala/com/spotify/scio/runners/dataflow/DataflowContext.scala

```diff
@@ -41,16 +41,22 @@ case object DataflowContext extends RunnerContext {
       .filesToStage(options, classLoader, localArtifacts, artifacts)
       .asJavaCollection
 
-    // Required for Kryo w/ Java 17
+    // Required for Kryo w/ Java 17+
     lazy val dataflowPipelineOpts = options.as(classOf[DataflowPipelineOptions])
-    if (
-      sys
-        .props("java.version")
-        .startsWith("17.") && dataflowPipelineOpts.getJdkAddOpenModules == null
-    ) {
-      dataflowPipelineOpts.setJdkAddOpenModules(
-        List("java.base/java.util=ALL-UNNAMED", "java.base/java.lang.invoke=ALL-UNNAMED").asJava
-      )
+    val javaVersionRe = "^(\\d+)\\..*".r
```

Comment:
This regex would give `1` for java 8 (version is set as `1.8....`).
Let's use the same def as for the build
```
System.getProperty("java.version").stripPrefix("1.").takeWhile(_.isDigit).toInt
```